onadata/apps/logger/management/commands/soft_delete_orphan_attachments.py

```diff
@@ -43,11 +63,58 @@ def handle(self, *args, **kwargs):
                 self.stdout.write(
                     f'Processing Instance object #{instance_id}{message}…'
                 )
-            soft_deleted_attachments = get_soft_deleted_attachments(instance)
+
+            if start_date and instance.date_created < datetime_from_str(
+                f'{start_date}T00:00:00 +0000'
+            ):
+                if verbosity > 1:
+                    message = '' if verbosity <= 1 else f' - {cpt}/{instances_count}'
+                    self.stdout.write(
+                        f'\tSkip Instance object #{instance_id}{message}. Too old'
+                    )
+                cpt += 1
+                continue
+
+            if not self._has_instance_been_edited(instance):
+                if verbosity > 1:
+                    message = '' if verbosity <= 1 else f' - {cpt}/{instances_count}'
+                    self.stdout.write(
+                        f'\tSkip Instance object #{instance_id}{message}. Not edited'
+                    )
+                cpt += 1
+                continue
+
+            try:
+                soft_deleted_attachments = get_soft_deleted_attachments(instance)
+            except Exception as e:
+                cpt += 1
+                if verbosity > 0:
+                    self.stderr.write(
+                        f'\tError for Instance object #{instance_id}: {str(e)}'
+                    )
+                continue
+
             for soft_deleted_attachment in soft_deleted_attachments:
                 pre_delete_attachment(
                     soft_deleted_attachment, only_update_counters=True
                 )
+            if verbosity > 1:
+                message = '' if verbosity <= 1 else f' - {cpt}/{instances_count}'
+                self.stdout.write(
+                    f'\tInstance object #{instance_id}{message} updated!'
+                )
+            cpt += 1
 
             cpt += 1
         self.stdout.write('Done!')
+
+    def _has_instance_been_edited(self, instance):
+        """
+        Consider instance as edited if it modification date is more or less 10
+        seconds apart
+        """
+        date_created_ts = instance.date_created.timestamp()
+        date_modified_ts = instance.date_modified.timestamp()
+        return not (
+            date_created_ts <= date_modified_ts <= date_created_ts + 10
+        )
```

Comment:
This could probably be a query, too. Here's my (untested) crack at it:

```code
queryset.filter(
    instance__date_created__lt=F('instance__date_modified'),
    instance__date_modified__lt=F('instance__date_created') + datettime.timedelta(seconds=10),
)
```

Reply:
See [comment](https://github.com/kobotoolbox/kobocat/pull/902#discussion_r1373000834) above